### PR TITLE
[PDMO-47] correct error display in encounters to inline

### PR DESCRIPTION
**Pull request by Ewanjiru**
# JIRA TICKET NAME:

PDMO-47 [Modify the display of error messages on the PDMO Encounters  Page](https://issues.openmrs.org/browse/PDMO-47)

# SUMMARY:

Make changes to the encounters page  so that the error will be shown on the affected field on the form instead of a toastr flash of the error message.

### CODE REVIEW

**Review by chibujax — COMMENTED** [review]
Looks good.

File: app/js/components/encounters/encounterProviders.jsx [hunk 1]
```diff
@@ -104,7 +104,10 @@ const Providers = (props) => {
             <div className="modal-body">
               <form>
                 <div className="form-group row">
-                  <label className="col-sm-6 col-form-label">Encounter Role </label>
+                  <label
```
Comment (CeciliaWanjiku) [inline app/js/components/encounters/encounterProviders.jsx]: check on this trailing white space

File: app/js/components/encounters/encounterProviders.jsx [hunk 2]
```diff
@@ -134,7 +137,11 @@ const Providers = (props) => {
                       <option value="" />
                       {
                         createProvidersArray.map(providerName => (
-                          <option key={providerName.uuid} value={providerName.uuid}>{providerName.display}</option>
+                          <option
```
Comment (CeciliaWanjiku) [inline app/js/components/encounters/encounterProviders.jsx]: this as well

File: app/js/components/encounters/encountersParent.jsx [hunk 3]
```diff
@@ -187,39 +206,51 @@ export default class Encounters extends React.Component {
...
       visit: editValues.visit,
-    }, 'post', `encounter/${this.state.encounterUuid}`)
+    }, 'post', `encounter/${encounter}`)
       .then((response) => {
+        this.setState({ editErrors: { error: '' } });
         if (response.error) {
-          this.fetchEncounterData(this.state.encounterUuid);
-          toastr.error(response.error.fieldErrors.encounterDatetime[0].message);
+          this.setState({ editErrors: { error: response.error.fieldErrors.encounterDatetime[0].message } });
+        } else {
```
Comment (CeciliaWanjiku) [inline app/js/components/encounters/encountersParent.jsx]: this as well

File: app/js/components/encounters/encountersParent.jsx [hunk 4]
```diff
@@ -338,20 +368,27 @@ export default class Encounters extends React.Component {
...
   handleObservationClick(observationUuid) {
+    const encounter = this.checkCurrentPatient();
+    let patientId = '';
+    const encounterId = '';
+    (encounter === this.state.newPatientEncounterUuid)
+      ? patientId = this.state.newPatientUuid
+      : patientId = this.state.patientUuid;
     this.props.router.push(
-      `/patient/${this.state.patientUuid}/encounter/${this.state.encounterUuid}/obs/${observationUuid}
+      `/patient/${patientId}/encounter/${encounter}/obs/${observationUuid}
```
Comment (CeciliaWanjiku) [inline app/js/components/encounters/encountersParent.jsx]: this as well

**Review by CeciliaWanjiku — COMMENTED** [review]
Looks good

**Comment by jacksono** [comment]
Good job

**Comment by dkayiwa** [comment]
@Ewanjiru can you look into these merge conflicts?

**Comment by Ewanjiru** [comment]
@dkayiwa  
Working on it.

**Comment by Ewanjiru** [comment]
@dkayiwa 
The merge conflicts have been resolved.

**Pull request merged**